### PR TITLE
Launchpad: Add Wrapper component

### LINKED ISSUE #77283: Create wrapper component for Launchpad
Make a generic “Launchpad” component that wraps the checklist/checklist item components and utilizes the connection to the API via `useLaunchpad` hook in data-stores.

Related to https://github.com/Automattic/wp-calypso/issues/77284 https://github.com/Automattic/wp-calypso/issues/77285

### PR DESCRIPTION
**Pull request by nelsonec87**
Closes https://github.com/Automattic/wp-calypso/issues/77283
Related to https://github.com/Automattic/wp-calypso/issues/77284 https://github.com/Automattic/wp-calypso/pull/77465

## Proposed Changes

* Added Wrapper component to the launchpad package (`<Launchpad />`)
* Moved checklist styles inside the package

## Testing 

* The functionality remains the same.
* Test if the Launchpad step still works correctly.
* Add the `Launchpad` component to a page and check if it's rendered correctly

```
import { Launchpad } from '@automattic/launchpad';
...
<Launchpad siteSlug={ siteSlug }></Launchpad>
```

image

### CODE REVIEW

File: client/my-sites/customer-home/main.jsx [hunk 1]
```diff
@@ -148,6 +149,7 @@ const Home = ( {
 			) : (
 				<>
 					<Primary cards={ layout.primary } />
+					<ChecklistWrapper siteSlug={ site?.slug }></ChecklistWrapper>
```
Comment (sixhours) [inline client/my-sites/customer-home/main.jsx]: I'd envisioned this being named `Launchpad` or similar, what do you think?

Comment (nelsonec87) [reply client/my-sites/customer-home/main.jsx]: It is, obviously, 100000% better!

**Comment by nelsonec87** [comment]
Rebased to solve some conflicts

**Comment by sixhours** [comment]
I did a side-by-side comparison of the Launchpad in Stepper with this PR and on production and noticed the "Launch your site" button looks off for this PR:

Production:

Screen Shot 2023-05-31 at 11 22 47 AM

This PR:

Screen Shot 2023-05-31 at 11 22 42 AM

**Comment by nelsonec87** [comment]
Rebasing to see if it clears the unrelated build errors

**Comment by valterlorran** [comment]
I think the progress is wrong. It says 4/5.
Screen Shot 2023-06-01 at 07 35 18

**Comment by nelsonec87** [comment]
> I think the progress is wrong. It says 4/5. Screen Shot 2023-06-01 at 07 35 18

Interesting @valterlorran , I'll try to find the inconsistency, but I don't think it's related to this PR. Does it show different progress on staging?

**Review by valterlorran — APPROVED** [review]
LGTM! You were right, not related to this PR. I think it is considering the Launch as 1 of the steps.

**Pull request merged**